src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2464,6 +2541,28 @@ private BoundExpression BindUnaryOperatorCore(CSharpSyntaxNode node, string oper
                 return ConstantValue.Create(newValue, resultType);
             }
 
+            try
+            {
+                newValue = FoldNativeIntegerOverflowingUnaryOperator(kind, value);
+            }
+            catch (OverflowException)
+            {
+                if (CheckOverflowAtCompileTime)
+                {
+                    Error(diagnostics, ErrorCode.ERR_CheckedOverflow, syntax);
+                    return ConstantValue.Bad;
+                }
+                else
+                {
+                    return null;
```

Comment:
>return null [](start = 20, length = 11)

This does not look correct.  Nowhere else in the language do we turn a constant expression into a non-constant expression because we don't know the result.  For the other cases we currently produce an "unspecified" constant result.  An expression specified to be a constant expression cannot fail to be constant-folded, though the result may be unspecified.  Please open an LDM issue about whether an expression that triggers this behavior should turn into a non-constant expression or result in an error.

Reply:
The `return null;` results in an error for `G1` below, but means the value `G2` will be calculated at runtime.
```C#
const nint F = int.MinValue;
const nint G1 = unchecked(-F); // CS0133: The expression must be constant
      nint G2 = unchecked(-F); // calculated at runtime
```

---
In reply to: [389839506](https://github.com/dotnet/roslyn/pull/42143#discussion_r389839506) [](ancestors = 389839506)

Reply:
Logged [https://github.com/dotnet/csharplang/issues/3259](https://github.com/dotnet/csharplang/issues/3259).

---
In reply to: [389855830](https://github.com/dotnet/roslyn/pull/42143#discussion_r389855830) [](ancestors = 389855830,389839506)